core/.native/src/main/scala/io/bullet/borer/internal/Unsafe.scala

```diff
@@ -0,0 +1,274 @@
+          throw new IllegalArgumentException(s"source Array[Byte] has illegal length: ${source.length}")
+        val ints = new Array[Int](source.length >> 2)
+        memcpy(ints.at(0), source.at(0), source.length.toSize.toUSize)
+        val target = new Array[Float](ints.length)
+        if (this.byteOrder != byteOrder)
+          @tailrec def rec(ix: Int): Array[Float] =
+            if (ix < target.length)
+              target(ix) = java.lang.Float.intBitsToFloat(java.lang.Integer.reverseBytes(ints(ix)))
+              rec(ix + 1)
+            else target
+          rec(0)
+        else
+          @tailrec def rec(ix: Int): Array[Float] =
+            if (ix < target.length)
+              target(ix) = java.lang.Float.intBitsToFloat(ints(ix))
+              rec(ix + 1)
+            else target
+          rec(0)
+      else Array.emptyFloatArray
+
+    def byteArrayToDoubleArray(source: Array[Byte], byteOrder: ByteOrder): Array[Double] =
+      if (source.length > 0)
+        if ((source.length & 7) != 0)
+          throw new IllegalArgumentException(s"source Array[Byte] has illegal length: ${source.length}")
+        val longs = new Array[Long](source.length >> 3)
+        memcpy(longs.at(0), source.at(0), source.length.toSize.toUSize)
+        val target = new Array[Double](longs.length)
+        if (this.byteOrder != byteOrder)
+          @tailrec def rec(ix: Int): Array[Double] =
+            if (ix < target.length)
+              target(ix) = java.lang.Double.longBitsToDouble(java.lang.Long.reverseBytes(longs(ix)))
+              rec(ix + 1)
+            else target
+          rec(0)
+        else
+          @tailrec def rec(ix: Int): Array[Double] =
+            if (ix < target.length)
+              target(ix) = java.lang.Double.longBitsToDouble(longs(ix))
+              rec(ix + 1)
+            else target
+          rec(0)
+      else Array.emptyDoubleArray
+
+  final class LittleEndianByteArrayAccess extends UnsafeByteArrayAccess(ByteOrder.LITTLE_ENDIAN):
+
+    def doubleByteBigEndian(byteArray: Array[Byte], ix: Int): Char =
+      Character.reverseBytes(_doubleByteBigEndian(byteArray, ix))
+
+    def quadByteBigEndian(byteArray: Array[Byte], ix: Int): Int =
+      Integer.reverseBytes(_quadByteBigEndian(byteArray, ix))
+
+    def octaByteBigEndian(byteArray: Array[Byte], ix: Int): Long =
+      JLong.reverseBytes(_octaByteBigEndian(byteArray, ix))
+
+    def setDoubleByteBigEndian(byteArray: Array[Byte], ix: Int, value: Char): Unit =
+      _setDoubleByteBigEndian(byteArray, ix, Character.reverseBytes(value))
+
+    def setQuadByteBigEndian(byteArray: Array[Byte], ix: Int, value: Int): Unit =
+      _setQuadByteBigEndian(byteArray, ix, Integer.reverseBytes(value))
+
+    def setOctaByteBigEndian(byteArray: Array[Byte], ix: Int, value: Long): Unit =
+      _setOctaByteBigEndian(byteArray, ix, JLong.reverseBytes(value))
+
+  final class BigEndianByteArrayAccess extends UnsafeByteArrayAccess(ByteOrder.LITTLE_ENDIAN):
```

Comment:
I am on my way home.
I will look deeper into it when I arrive.

Reply:
Yeah, that's wrong there as well.
This bug probably never got triggered so far since "direct JSON parsing" is so far only available on little endian platforms. This should be improved...

Reply:
(not in this PR though!)

Reply:
I looked a bit more into this topic and, to be honest, big endian appears to be pretty much dead regarding the commercial importance. There are almost no relevant big-endian-only platforms produced anymore.
Everything out there is either directly little endian oder at least bi-endian, which ends up being configured to almost exclusively run in little-endian mode.

So, since _borer_ should work fine on big endian systems as it is I'd rather spare the work for any big-endian optimizations. In all likelihood they simply won't ever see production use anywhere...

Reply:
Does that mean that we leave the big endian support untested?

Reply:
If yes, what else needs to be done to get this merged?

Reply:
> Does that mean that we leave the big endian support untested?

I've just created https://github.com/sirthias/borer/issues/731 for this.
Nothing big-endian related required in this PR.
👍

Reply:
> If yes, what else needs to be done to get this merged?

Give me a bit more time to look at this PR and then I think we should be fine to merge.